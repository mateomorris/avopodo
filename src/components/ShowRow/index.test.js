import React from 'react';
import { Text } from 'react-native';
import { ShowRow } from './index';
import renderer from 'react-test-renderer';

it('renders ShowRow correctly', () => {
  const tree = renderer
    .create(
        <ShowRow 
          index={0}
          item={{
            description: `The economy explained. Imagine you could call up a friend and say, "Meet me at the bar and tell me what's going on with the economy." Now imagine that's actually a fun evening.`,
            id: `8cb941141a7c434d945397cfd8b12e58`,
            image: `https://d3sv2eduhewoas.cloudfront.net/channel/image/4ffedff25d474163b1a8511a6a26b76a.jpg`,
            itunesId: 290783428,
            publisher: `NPR`,
            title: `Planet Money`
          }}
          subscribed={true}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
