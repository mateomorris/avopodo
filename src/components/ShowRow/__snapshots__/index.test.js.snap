// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ShowRow correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "#DDD",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "height": 100,
      "opacity": 1,
      "overflow": "hidden",
      "paddingBottom": 15,
      "paddingTop": 15,
    }
  }
>
  <View
    style={
      Object {
        "marginRight": 10,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://d3sv2eduhewoas.cloudfront.net/channel/image/4ffedff25d474163b1a8511a6a26b76a.jpg",
        }
      }
      style={
        Object {
          "backgroundColor": "#EEEEEE",
          "borderRadius": 5,
          "height": 75,
          "width": 75,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "fontSize": 15,
          "fontWeight": "700",
        }
      }
    >
      Planet Money
    </Text>
    <Text
      ellipsizeMode="tail"
      numberOfLines={4}
      style={
        Object {
          "color": "gray",
          "fontSize": 12,
        }
      }
    >
      The economy explained. Imagine you could call up a friend and say, "Meet me at the bar and tell me what's going on with the economy." Now imagine that's actually a fun evening.
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/bookmark-black.png",
        }
      }
      style={
        Object {
          "height": 25,
          "width": 25,
        }
      }
    />
  </View>
</View>
`;
